File: extract_colors.py
from PIL import Image
from collections import Counter

def get_dominant_colors(image_path, num_colors=5):
    try:
        image = Image.open(image_path)
        image = image.convert("RGB")
        image = image.resize((150, 150))  # Resize to speed up processing
        pixels = list(image.getdata())
        
        # Filter out near-white and near-black pixels to find the colors
        filtered_pixels = [
            p for p in pixels 
            if not (p[0] > 240 and p[1] > 240 and p[2] > 240) and # Not white
               not (p[0] < 15 and p[1] < 15 and p[2] < 15)        # Not black
        ]
        
        counts = Counter(filtered_pixels)
        dominant = counts.most_common(num_colors)
        
        print(f"Dominant colors in {image_path}:")
        for color, count in dominant:
            hex_color = '#{:02x}{:02x}{:02x}'.format(color[0], color[1], color[2])
            print(f"- {hex_color} (RGB: {color}) - Count: {count}")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    get_dominant_colors("public/brand/logo.png")

File: extract_colors.mjs
import Jimp from "jimp";

async function main() {
    try {
        const image = await Jimp.read("public/brand/logo.png");
        const width = image.bitmap.width;
        const height = image.bitmap.height;
        const colorCounts = {};

        image.scan(0, 0, width, height, function (x, y, idx) {
            const red = this.bitmap.data[idx + 0];
            const green = this.bitmap.data[idx + 1];
            const blue = this.bitmap.data[idx + 2];
            const alpha = this.bitmap.data[idx + 3];

            if (alpha < 255) return; // Ignore transparent

            // Ignore white-ish and black-ish
            if (red > 240 && green > 240 && blue > 240) return;
            if (red < 20 && green < 20 && blue < 20) return;

            const hex = "#" + ((1 << 24) + (red << 16) + (green << 8) + blue).toString(16).slice(1);

            colorCounts[hex] = (colorCounts[hex] || 0) + 1;
        });

        const sortedColors = Object.entries(colorCounts)
            .sort((a, b) => b[1] - a[1])
            .slice(0, 5);

        console.log("Dominant colors:");
        sortedColors.forEach(([color, count]) => {
            console.log(`${color}: ${count}`);
        });

    } catch (err) {
        console.error(err);
    }
}

main();
